cloud/endagaweb/views/user.py

```diff
@@ -123,13 +147,28 @@ def change_password(request):
     required_params = ('old_password', 'new_password1', 'new_password2')
     if not all([param in request.POST for param in required_params]):
         return HttpResponseBadRequest()
-    # Validate
-    redirect_url = '/dashboard/profile'
+    # Validate url for redirect
+    if urlparse.urlparse(request.META['HTTP_REFERER']
+                         ).path != '/dashboard/profile':
+        redirect_url = '/password/change'
+    else:
+        redirect_url = '/dashboard/profile'
     if not request.user.check_password(request.POST['old_password']):
         text = 'Error: old password is incorrect.'
         tags = 'password alert alert-danger'
         messages.error(request, text, extra_tags=tags)
         return redirect(redirect_url)
+    if not validate_password_strength(request.POST['new_password1']):
+        text = 'Error: password must contain at least 8 characters,contains ' \
+               'alphanumeric and one special character.'
+        tags = 'password alert alert-danger'
+        messages.error(request, text, extra_tags=tags)
+        return redirect(redirect_url)
+    if request.POST['old_password'] == request.POST['new_password1']:
```

Comment:
You should be doing this validation in the PasswordForm using the `clean_new_password1`  method. See this example: https://docs.djangoproject.com/en/1.8/_modules/django/contrib/auth/forms/#SetPasswordForm